src/lib/vm/core.rs

```diff
@@ -832,9 +832,36 @@ impl VM {
             }
             Primitive::InstVarNamed => unimplemented!(),
             Primitive::InvokeOnWith => todo!(),
-            Primitive::IsDigits => unimplemented!(),
-            Primitive::IsLetters => unimplemented!(),
-            Primitive::IsWhiteSpace => unimplemented!(),
+            Primitive::IsDigits => {
+                let str_val = stry!(rcv.downcast::<String_>(self));
+                let s = str_val.as_str();
+                if s.len() > 0 && s.chars().all(|x| x.is_ascii_digit()) {
```

Comment:
Maybe using `str::is_empty` can be more idiomatic here ?
```suggestion
                if !s.is_empty() && s.chars().all(|x| x.is_ascii_digit()) {
```

Reply:
Good point!

Reply:
Fixed in https://github.com/softdevteam/yksom/pull/153/commits/62bf7067e48cfc834f1061d675c90ab1397c4825.